### PR TITLE
Feature/20 allow admin to edit an existing article

### PR DESCRIPTION
**Pull request by pro100proger**

### CODE REVIEW

File: src/main/java/com/softserve/edu/sporthubujp/controller/ArticleController.java [hunk 1]
```diff
@@ -30,4 +31,10 @@ public ResponseEntity<Void> deleteArticle(@PathVariable("id") String articleId)
         articleService.deleteArticleById(articleId);
         return ResponseEntity.status(HttpStatus.OK).build();
     }
+    @PutMapping(path = "/{id}")
+    public ResponseEntity<Article> updateArticle(@RequestBody Article newArticle, @PathVariable("id") String id)
+    {
```
Comment (AndriyBarskyi) [inline src/main/java/com/softserve/edu/sporthubujp/controller/ArticleController.java:36]: Code style

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,8 +1,8 @@
 spring.datasource.username=postgres
-spring.datasource.password=Lv_2022_06
+spring.datasource.password=this1pass
```
Comment (AndriyBarskyi) [inline src/main/resources/application.properties:2]: Do not change properties

**Review by AndriyBarskyi — CHANGES_REQUESTED** [review]

**Review by AndriyBarskyi — APPROVED** [review]

**Review by ostartt — APPROVED** [review]

File: src/main/java/com/softserve/edu/sporthubujp/entity/Article.java [hunk 3]
```diff
@@ -12,7 +12,7 @@
 @Entity
 @Data
 @EntityListeners(AuditingEntityListener.class)
-@Table(name="ARTICLE")
+@Table(name="ARTICLES")
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/entity/Article.java:15]: Change in liquibase also

**Review by yuremboo — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserve/edu/sporthubujp/entity/Article.java [hunk 4]
```diff
@@ -24,6 +24,14 @@ public class Article {
     private String title ;
     @Column(columnDefinition="text", name = "text", nullable = false, unique = false)
     private String text ;
+    @Column(name = "caption", length = 255, nullable = false, unique = false)
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/entity/Article.java:27]: these changes should also go to liquibase

File: src/main/java/com/softserve/edu/sporthubujp/service/impl/ArticleServiceImpl.java [hunk 5]
```diff
@@ -37,4 +36,24 @@ public void deleteArticleById(String id)
         }
         articleRepository.deleteById(id);
     }
+
+    public Article updateArticle(Article newArticle, String id)
+    {
+        return articleRepository.findById(id)
+            .map(article -> {
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/service/impl/ArticleServiceImpl.java:43]: you can use MapStruct like:

`void update(@MappingTarger Article articleFromDb, Article newArticle);`

**Review by yuremboo — CHANGES_REQUESTED** [review]

**Review by ostartt — APPROVED** [review]

**Review by AndriyBarskyi — APPROVED** [review]

File: src/main/java/com/softserve/edu/sporthubujp/controller/ArticleController.java [hunk 6]
```diff
@@ -53,6 +54,15 @@ public ResponseEntity<Void> deleteArticle(@PathVariable("id") String articleId)
         return ResponseEntity.status(HttpStatus.OK).build();
     }
 
+
+    @PutMapping(path = "/{id}")
+    @PreAuthorize("hasAnyAuthority('USER', 'ADMIN')")
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/controller/ArticleController.java:59]: only ADMIN can update article

**Review by yuremboo — CHANGES_REQUESTED** [review]

**Review by yuremboo — APPROVED** [review]

**Pull request merged**